lib/game_of_life_web/live/universe_live.ex

```diff
@@ -38,7 +38,22 @@ defmodule GameOfLifeWeb.UniverseLive do
 
   def handle_info(:tick, socket), do: {:noreply, put_cells(socket, &Universe.tick/1)}
 
+  def handle_event("update_speed", %{"speed" => speed}, socket) do
+    {:noreply, set_timer(socket, String.to_integer(speed))}
+  end
+
   defp put_cells(socket, f), do: assign(socket, cells: f.(socket.assigns.universe))
 
   defp rand_bytes, do: :crypto.strong_rand_bytes(16)
+
+  defp set_timer(socket, speed) do
+    if Map.has_key?(socket.assigns, :timer_ref) do
+      :timer.cancel(socket.assigns.timer_ref)
+    end
+
+    # `send_interval` needs an Integer, not a Float
+    {:ok, tref} = :timer.send_interval(trunc(2000 / speed), self(), :tick)
+
+    assign(socket, timer_ref: tref)
```

Comment:
I wasn't sure if I should store the `speed` itself on the socket - I was concerned that storing both speed AND reference to the timer would open up the possibility that they get out of sync somehow, so the timer ref is currently the single source of truth

Reply:
not 100% sure on this, but I think you want to store the speed as well since that is what the slider's value is bound to.  I'll look a little bit more

Reply:
That makes sense actually, so instead of `value="6"` in the template it would be `value=@speed`?

Reply:
I was looking at some of the example LiveView apps and I don't think we want to actually store the `timer`.  The method other apps us is `Process.send_after/3`.  This allows the app to continue to schedule just the next render, rather than having a persistent process running.  Ex:

https://github.com/chrismccord/phoenix_live_view_example/blob/master/lib/demo_web/live/rainbow_live.ex

Reply:
Nice, thanks! I'll model this request after that